internal/cmd/rootcmd/env.go

```diff
@@ -14,14 +14,15 @@ const envKeyPrefix = "CONTROLLER_"
 // bindEnvVars, for each flag defined on `cmd` (local or parent persistent), looks up the corresponding environment
 // variable and (if the flag is unset) takes that environment variable value as the flag value.
 func bindEnvVars(cmd *cobra.Command, _ []string) (err error) {
+	var envKey string
 	defer func() {
 		if r := recover(); r != nil {
-			err = fmt.Errorf("environment binding failed: %v", r)
+			err = fmt.Errorf("environment binding failed: environment variable %s invalid value: %v", envKey, r)
```

Comment:
A potential problem with this approach is that it kind of stutters when the message is printed, e.g.:

`environment variable CONTROLLER_VALIDATION_TEST invalid value: bad value for var type testvar`

`invalid value: bad value` is somewhat redundant. How do you feel about a change?

```suggestion
			err = fmt.Errorf("environment binding failed for variable %s: %v", envKey, r)
```

The resulting output for the previous example would then be:

`environment binding failed for variable CONTROLLER_VALIDATION_TEST: bad value for var type testvar`

I think this might read a little better, let me know what you think?

Reply:
yes, sound's good to me. will commit the changes ✌🏻